### PR TITLE
Tools 2247 support boolean bin and udf arg values

### PR DESCRIPTION
**Pull request by jdogmcsteezy**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,9 +17,6 @@ ifeq ($(OS),Darwin)
   CFLAGS += -I/usr/local/include -D_DARWIN_UNLIMITED_SELECT
   ifneq ($(wildcard /usr/local/opt/openssl/include),)
     CFLAGS += -I/usr/local/opt/openssl/include
-    ifeq ($(OPENSSL_STATIC_PATH),)
```
Comment (jdogmcsteezy) [inline Makefile:26]: redundant to like 37

File: src/main/asql_parser.c [hunk 2]
```diff
@@ -724,6 +724,18 @@ parse_value(char* s, asql_value* value)
 		return 0;
 	}
 
+	if (strcasecmp(s, "true") == 0) {
```
Comment (jdogmcsteezy) [inline src/main/asql_parser.c:727]: parses input string into asql_value type

File: src/main/asql_value.c [hunk 3]
```diff
@@ -130,6 +134,19 @@ asql_set_args(as_error* err, as_vector* udfargs, as_arraylist* arglist)
 				}
 				break;
 			}
+			case AS_BOOLEAN:
+			{
+				as_val* bol;
+
+				if (value->u.bol) {
+					bol = as_boolean_toval(&as_true);
```
Comment (jdogmcsteezy) [inline src/main/asql_value.c:142]: converts an intermediate asql_value into an as_val which is what is provided as udf args

File: src/main/asql_value.c [hunk 4]
```diff
@@ -303,6 +320,42 @@ asql_parse_value_as(char* s, asql_value* value, asql_value_type_t vtype)
...
+			int64_t val = strtoll(s, &endptr, 0);
+
+			if (*endptr == 0 || *endptr == s0)
+			{
+				value->type = AS_BOOLEAN;
+				value->u.bol = (val == 0) ? false : true;
+			}
+			else
+			{
+				fprintf(stdout, "Error!  Cannot cast \"%s\" to float!\n", s);
```
Comment (dwelch-spike) [inline src/main/asql_value.c:354]:
```suggestion
				fprintf(stdout, "Error!  Cannot cast \"%s\" to bool!\n", s);
```

**Review by dwelch-spike — CHANGES_REQUESTED** [review]
Error message suggestion. Looks good otherwise

**Pull request merged**